### ISSUE TITLE
Can you update the tests with the latest versions of the software?

### ISSUE THREAD

**Issue by bozhodimitrov** [svanoort/python-client-benchmarks#14]
Hello @svanoort,
I saw your benchmark and decided to do my own research on this topic.
I'm using the latest versions of python 3.5.2, requests 2.13.0 and pycurl 7.43.0
And according to my tests, requests is only ~4% slower with connection reuse (HTTP/HTTPS).
Didn't test the CPU times, because of the observed big difference in your and my results.

By the way - my tests return the response body, response headers and status code (the three most important things that are needed for the HTTP communication).
In your benchmark, you should always return the status code and parse the headers -provide them as variables. This way we can have better apples to apples comparison since requests does this in any case.

Can you please update your benchmark, because it seems outdated and a lot of people might get it wrong.
Thanks in advance for your time!

**Comment by svanoort** [comment]
Hi @int3l,
The key thing to remember with benchmarks is that they depend on the environment and configuration they are run in.  When I tested, it was in an environment designed to simulate real-world use -- AWS instances, testing both in local loopback and from instance to instance.  This insured there were no sources of external interference in measurement, and also verified behavior when communicating across the network.  When taking such measurements, it is absolutely CRITICAL to do them in a clean environment, and to include BOTH cpu time and wall time; wall time helps to estimate real-world latency, while CPU time is useful for scalability.  Additionally I tracked system-level stats in loopback [from vmstat](https://github.com/svanoort/python-client-benchmarks/blob/master/new-aws-results/vmstat-loopback-https.txt) to ensure that there were no additional bottlenecks at work before doing tests.

It took quite a while to run the measurements, analyze, and then format the data into graphs, so I did not have time to cross-test against different language and library versions.  Obviously the results will vary depending on the specific versions and the hardware used. This does **not** mean test results are "outdated" or "wrong" -- it just means that performance differs with version.

Given I have a demanding full-time job working with other senior-level engineers, I do not have further time to retest this & regenerate the analysis simply because you managed to obtain different numbers with an unspecified test methodology.  However, you are more than welcome to use the documented test methodology & included tools to run your own benchmarks -- if you include both Python 2 and 3, I'll be happy to link it as a further set of results.

**TL;DR:** Getting good data is not as simple as you'd think, and there are multiple variables.  I've tested several combinations here with a high degree of rigor.  You're more than welcome to try some additional variables and if the benchmark methodology is sound, I'll link the additional info.

Regards,
Sam

**Comment by svanoort** [comment]
Closing this out, since it's been responded to, and doesn't require additional action.
